buck-worker/Main.hs

```diff
@@ -58,11 +77,31 @@ compileAndReadAbiHash args target = do
         Just AbiHash {path, hash = showAbiHash hsc_env artifacts.iface}
     pure CompileResult {artifacts, abiHash}
 
+startJob ::
+  MVar WorkerStatus ->
+  IO ()
+startJob var =
+  modifyMVar_ var \ WorkerStatus {active} -> do
+    let new = active + 1
+    dbg ("Starting job, now " ++ show new ++ " active")
+    pure WorkerStatus {active = new}
+
+finishJob ::
+  MVar WorkerStatus ->
+  a ->
+  IO ()
+finishJob var _ = do
+  modifyMVar_ var \ WorkerStatus {active} -> do
+    let new = active - 1
+    dbg ("Finishing job, now " ++ show new ++ " active")
+    pure WorkerStatus {active = new}
+
 executeHandler ::
+  MVar WorkerStatus ->
   MVar Cache ->
   ServerRequest 'Normal ExecuteCommand ExecuteResponse ->
   IO (ServerResponse 'Normal ExecuteResponse)
-executeHandler cache (ServerNormalRequest _ ExecuteCommand {executeCommandArgv, executeCommandEnv}) = do
+executeHandler status cache (ServerNormalRequest _ ExecuteCommand {executeCommandArgv, executeCommandEnv}) = do
   when False do
```

Comment:
This is better to be (1) removed if it is a temporary debug, (2) commented out with a note or (3) `when debugMode` and define `debugMode = False` somewhere at top level. 
Just caught on my eyes. not necessary for this PR.